crates/solidity/inputs/schema/grammar/05-expressions/03-numbers/productions.yml

```diff
@@ -2,43 +2,78 @@
 
 - name: "NumericLiteral"
   kind: "Parser"
-  unversioned:
-    sequence:
-      - choice:
-          - reference: "HexLiteral"
-          - reference: "DecimalLiteral"
-      - optional:
-          reference: "NumberUnit"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - reference: "HexLiteral"
+            - reference: "DecimalLiteral"
+        - optional:
+            reference: "NumberUnit"
+    0.5.0:
+      sequence:
+        - reference: "DecimalLiteral"
+        - optional:
+            reference: "NumberUnit"
 
 - name: "HexLiteral"
   kind: "Scanner"
-  unversioned:
-    sequence:
-      - terminal: "0x"
-      - oneOrMore:
-          reference: "HexCharacter"
-      - zeroOrMore:
-          sequence:
-            - terminal: "_"
-            - oneOrMore:
-                reference: "HexCharacter"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - terminal: "0x"
+            - terminal: "0X"
+        - oneOrMore:
+            reference: "HexCharacter"
+        - zeroOrMore:
+            sequence:
+              - terminal: "_"
+              - oneOrMore:
+                  reference: "HexCharacter"
+    0.5.0:
+      sequence:
+        - terminal: "0x"
+        - oneOrMore:
+            reference: "HexCharacter"
+        - zeroOrMore:
+            sequence:
+              - terminal: "_"
+              - oneOrMore:
+                  reference: "HexCharacter"
 
 - name: "DecimalLiteral"
   kind: "Scanner"
-  unversioned:
-    sequence:
-      - choice:
-          - sequence:
-              - reference: "DecimalNumber"
-              - optional:
-                  sequence:
-                    - terminal: "."
-                    - reference: "DecimalNumber"
-          - sequence:
-              - terminal: "."
-              - reference: "DecimalNumber"
-      - optional:
-          reference: "DecimalExponent"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - sequence:
+                - reference: "DecimalNumber"
+                - optional:
+                    sequence:
+                      - terminal: "."
+                      - optional:
```

Comment:
Not sure why the second `DecimalNumber` became optional? this caused an existing test to pass, while it should have failed.

Reply:
Because `1.` is a valid float number before `0.5.0`